### PR TITLE
Adding Limited Xarray Support in MHKiT

### PR DESCRIPTION
**Pull request by Alex-McVey**
Current idea behind implementation is:
- for all functions that get data and return it to the user, there will be a flag for xarray
- for all functions that take data as an input, there will be a decorator that converts data

### CODE REVIEW

**Review by ssolson — COMMENTED** [review]
Hey Alex real quick comment on how to format docstrings for MHKiT. I am starting to play around with this and hope to have more for you later this week.

**Comment by Alex-McVey** [comment]
This PR is ready for review. As discussed I reset the changes in wave/resources.py so that the only changes in this PR are to functions that fetch data for the users.

**Comment by ssolson** [comment]
@Alex-McVey I opened a PR on your branch to merge the changes so the tests can run for this PR.

**Comment by ssolson** [comment]
Hey Alex my PR is still open on your branch. I pushed some changes that fix some bugs around the import statements of utils. I also got the branch up to date with the current Development branch. If you review and merge my changes we should be able to run the tests here to make sure everything is passing. I will be out of the office until the 6th of September but I'll catch back up then.

**Review by Alex-McVey — COMMENTED** [review]
@ssolson your changes look good to me. I can't quite tell from this view if you changed the docstrings or if I still need to do it.

**Comment by ssolson** [comment]
@Alex-McVey I am not seeing the tests for this PR. Can you push up the tests you wrote? I think we need to add the functionality as an example to at least one of the examples that we can point people to on the PRs use.

File: mhkit/wave/resource.py [hunk 1]
```diff
@@ -9,7 +9,6 @@ def elevation_spectrum(eta, sample_rate, nnft, window='hann',
     detrend=True, noverlap=None):
     """
     Calculates the wave energy spectrum from wave elevation time-series
-
```
[the remark this answers is not in the document]
Comment (akeeste) [reply mhkit/wave/resource.py:12]: Can you restore this file to the previous version? Nothing functional was changed in ``resource.py``

**Review by akeeste — COMMENTED** [review]
Hi @Alex-McVey 

I got around to reviewing this PR. I went through the first round of my review for understanding and commented on a few things. I still need to go through and do a more functional check. Can you:
- [x] Update the ``return`` in update function doc strings to describe the output as a "pandas DataFrame or xarray Dataset". I meant to add suggestions as I went, but currently GitHub does not allow suggestions on most unchanged lines
- [ ] Pull in the most recent version of Develop, especially the testing restructure. This will help my next review go more easily

Adam

**Comment by akeeste** [comment]
This feature is superseded by #221 will continue there

**Pull request closed without merging**